libraries/artifactory_gavc_downloader.rb

```diff
@@ -31,7 +32,7 @@ def find_newest_version_for_coordinates(group: raise('group is required'),
                                           artifact: raise('artifact is required'),
                                           version: raise('version is required'),
                                           repos: raise('repo(s) are required'))
-    return version unless wildcarded_version?(version)
+    return version if (!wildcarded_version?(version) && !snapshot_version?(version))
```

Comment:
The problem was that the function find_newest_version_for_coordinates don't lookup for a newest version if version = "1.0.0-SNAPSHOT", the function only lookup if the version is a wildcard version.
But we need the function for snapshot versions too, because we have 1.0.0-SNAPSHOT version with attached timestamp, so the function must find the last snapshot version. This wasn't possible before.